Framework/Algorithms/test/CorelliPowderCalibrationLoadTest.h

```diff
@@ -0,0 +1,124 @@
+    TS_ASSERT_EQUALS(corelliLoader.name(), "CorelliPowderCalibrationLoad");
+  }
+
+  void testInit() {
+    CorelliPowderCalibrationLoad corelliLoader;
+    corelliLoader.initialize();
+    TS_ASSERT(corelliLoader.isInitialized());
+  }
+
+  void testValidateWSType() {
+    // generate a mock workspace with wrong instrument name
+    IAlgorithm_sptr lei =
+        AlgorithmFactory::Instance().create("LoadEmptyInstrument", 1);
+    lei->initialize();
+    lei->setPropertyValue("Filename", "NOW4_Definition.xml");
+    lei->setPropertyValue("OutputWorkspace", "wrongTypeWs");
+    lei->setPropertyValue("MakeEventWorkspace", "1");
+    lei->execute();
+    EventWorkspace_sptr ws =
+        AnalysisDataService::Instance().retrieveWS<EventWorkspace>(
+            "wrongTypeWs");
+    ws->mutableRun().addProperty<std::string>("start_time",
+                                              "2020-11-17T12:57:17", "", true);
+
+    // setup alg
+    CorelliPowderCalibrationLoad alg;
+    alg.initialize();
+    alg.setPropertyValue("InputWorkspace", "wrongTypeWs");
+    alg.setPropertyValue("DatabaseDir", ".");
+    alg.setPropertyValue("OutputWorkspace", "outWS");
+
+    // ensure that an exception is thrown here
+    TS_ASSERT_THROWS(alg.execute(), const std::runtime_error &);
+  }
+
+  void testValidateWSTime() {
+    // generate a mock workspace with correct instrument name
+    IAlgorithm_sptr lei =
+        AlgorithmFactory::Instance().create("LoadEmptyInstrument", 1);
+    lei->initialize();
+    lei->setPropertyValue("Filename", "CORELLI_Definition.xml");
+    lei->setPropertyValue("OutputWorkspace", "correctTypeWs");
+    lei->setPropertyValue("MakeEventWorkspace", "1");
+    lei->execute();
+    EventWorkspace_sptr ws =
+        AnalysisDataService::Instance().retrieveWS<EventWorkspace>(
+            "correctTypeWs");
+
+    // setup alg
+    CorelliPowderCalibrationLoad alg;
+    alg.initialize();
+    alg.setPropertyValue("InputWorkspace", "correctTypeWs");
+    alg.setPropertyValue("DatabaseDir", ".");
+    alg.setPropertyValue("OutputWorkspace", "outWS");
+
+    // ensure that an exception is thrown here
+    TS_ASSERT_THROWS(alg.execute(), const std::runtime_error &);
+  }
+
+  void testExec() {
+    // generate a mock workspace with correct instrument name
+    IAlgorithm_sptr lei =
+        AlgorithmFactory::Instance().create("LoadEmptyInstrument", 1);
+    lei->initialize();
+    lei->setPropertyValue("Filename", "CORELLI_Definition.xml");
+    lei->setPropertyValue("OutputWorkspace", "correctTypeWs");
+    lei->setPropertyValue("MakeEventWorkspace", "1");
+    lei->execute();
+    EventWorkspace_sptr ws =
+        AnalysisDataService::Instance().retrieveWS<EventWorkspace>(
+            "correctTypeWs");
+
+    // add a starting time to log
+    ws->mutableRun().addProperty<std::string>("start_time",
+                                              "2020-11-17T12:57:17", "", true);
+
+    // setup alg
```

Comment:
How about creating a small calibration table and save it to file with saveascii in a directory you can later use as the database?

Reply:
I have to say I do no support generating artifacts without a clear way to clean them afterwards, especially after hearing so many horror stories about how some leftovers from previous build mess up all subsequent builds on a build server.

Reply:
The teardown method should remove the file you create. Why's this insecure?

Reply:
Could you point me to the location where the teardown happens?

Reply:
Found a TestHelper class, `ScopedFile`, that can generate a tmp file on the fly and remove the file in its deconstructor.  Implementing it now.

Reply:
There's nothing new under the Mantid sun.